File: shellsort.go
package main
import "fmt"
var n int
var array []int
func search(array []int,n int){
	var gap,i,j,temp int
	for gap=n/2;gap>0;gap/=2{
		for i=gap;i<n;i++{
			temp=array[i]
			for j=i;j>=gap&&array[j-gap]>temp;j-=gap{
				array[j]=array[j-gap]
			}
			array[j]=temp
		}
	}
}
func main(){
	fmt.Printf("Enter the size of the array: ")
	fmt.Scanf("%d",&n)
	array=make([]int,n)
	fmt.Printf("Enter the element in the array:\n")
	for i:=0;i<n;i++{
		fmt.Scanf("%d ",&array[i])
	}
	search(array,n)
	fmt.Printf("\nSorted array:\n")
	for i:=0;i<n;i++{
		fmt.Printf("%d ",array[i])
	}
}

File: shellSort.js
function shellSort(array)
{
  let size=array.length,temp,j;
  //we will first reduce the array to half and assign it to the gap
  for(let gap=Math.floor(size/2) ; gap>0; gap=Math.floor(gap/2))
    {
      //then we will reduce the gap to one
      for(let i=gap; i<size;i+=1)
        {
          //a[i] the element after the gap will be store in temp
          //then we will compare it.
          temp=array[i];
          //shift the arrays by comparing it and until correct            location is not found
          for(j=i; j >= gap && arr[j-gap] > temp ; j-=gap)
                      array[j]=array[j-gap];
          array[j]=temp; 
       // console.log(j)      
        }
      
    }
  return 0;
}
//test 
let size=prompt("enter size");
let arr=[];
for(let i=0;i<size;i++)
  {
   arr[i]=prompt("Enter array");
  }
shellSort(arr);
console.log(arr);